compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/parser/BLangNodeTransformer.java

```diff
@@ -1095,7 +1095,7 @@ private BLangNode createService(Node serviceNode, IdentifierToken serviceNameNod
         // 1) Define type nodeDefinition for service type.
         BLangClassDefinition classDef = (BLangClassDefinition) TreeBuilder.createClassDefNode();
         BLangIdentifier serviceTypeID = createIdentifier(identifierPos, serviceTypeName);
-        serviceTypeID.pos = pos;
+        serviceTypeID.pos = symTable.builtinPos;
```

Comment:
Won't this result in some incorrect line-numbers for server declarations?

Reply:
Hmm.. Good point. Will check this and see. It looks like we are not skipping class defs generated for services. Will have to filter these in another way if that's the case.